util_test.go

```diff
@@ -0,0 +1,41 @@
+package dogma_test
+
+import (
+	"testing"
+)
+
+// messageWithPointerReceivers is an implementation of [Message] that uses
+// pointer receivers, used to test reflection code that has pointer-specific
+// logic.
```

Comment:
Nitpick: This feels like it could be two sentences or needs some extra words?
```suggestion
// messageWithPointerReceivers is an implementation of [Message] that uses
// pointer receivers, which is used to test reflection code that has pointer-specific
// logic.
```